tests/utils/esx/esxcli.go

```diff
@@ -77,3 +78,39 @@ func WaitForExpectedState(fn getVMPowerStatus, vmName, expectedState string) boo
 	log.Printf("Timed out to poll status\n")
 	return false
 }
+
+// IsVMBackOnline checks if VM is fully back online after being restarted
+func IsVMBackOnline(vmIP, vmName string) bool {
+	isStatusOnline := WaitForExpectedState(GetVMPowerState, vmName, properties.PowerOnState)
+	if !isStatusOnline {
+		log.Printf("VM %s is not in PoweredOn state", vmName)
+		return false
+	}
+
+	isVDVSRunning := isVDVSRunning(vmIP)
+	if !isVDVSRunning {
+		log.Printf("vDVS is not running on VM [%s]. Please check if the previous VM IP [%s] has changed.", vmName, vmIP)
+		return false
+	}
+
+	return true
+}
+
+// isVDVSRunning checks if vDVS is running. This util can be useful
+// in scenarios where vm is powered-on and user wants to find out if
+// vm is fully up to be able to run docker volume commands.
+func isVDVSRunning(ip string) bool {
```

Comment:
You are correct that we are checking the vDVS **plugin** (not the driver) which is running on VM. As for why this method is not exported, I have already explained the reason above. Original proposal was valid because at that time this util was invoked by multiple test cases directly, which is not valid anymore.

Anyway, since you both have the same comments, I will extract this util to a separate file.